File: CS/CircularProgressBar/Converters/ThicknessToOffsetConverter.cs
using System;
using System.Windows.Data;
using System.Windows.Markup;

namespace CircularProgressBar.Converters
{
    public class ThicknessToOffsetConverter : MarkupExtension, IValueConverter
    {
        public override object ProvideValue(IServiceProvider serviceProvider)
        {
            return this;
        }

        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            int factor = 1;
            if (parameter is null && value is null)
                return 0;

            var param = (string)parameter;
            var val = (int)value;
            if(!int.TryParse(param, out factor))
                return 0;

            return - (val + 7) * (factor - 1);
        }

        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            throw new NotImplementedException();
        }
    }
}

File: VB/CircularProgressBar/Converters/ThicknessToOffsetConverter.vb
Imports System
Imports System.Windows.Data
Imports System.Windows.Markup

Namespace CircularProgressBar.Converters

    Public Class ThicknessToOffsetConverter
        Inherits MarkupExtension
        Implements IValueConverter

        Public Overrides Function ProvideValue(ByVal serviceProvider As IServiceProvider) As Object
            Return Me
        End Function

        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As Globalization.CultureInfo) As Object Implements IValueConverter.Convert
            Dim factor As Integer = 1
            If parameter Is Nothing AndAlso value Is Nothing Then Return 0
            Dim param = CStr(parameter)
            Dim val = CInt(value)
            If Not Integer.TryParse(param, factor) Then Return 0
            Return -(val + 7) * (factor - 1)
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function
    End Class
End Namespace
